### PR TITLE
[Faucet] Add generated TS client

### PR DESCRIPTION
**Pull request by banool**
### Description
We had this in aptos-tap, this is mostly just copying it across. Other changes include:
- Change the name of the package to `aptos-labs/aptos-faucet-client`. I have deprecated `aptos-tap-client` in npmjs and added a deprecation notice: https://www.npmjs.com/package/aptos-tap-client.
- Change the license to Apache 2.0.
- Change to pnpm from yarn.
- Fix the tests.
- Use the latest version of the client code generator.

### Test Plan
From `crates/aptos-faucet/ts-client`:
```
cargo run -p aptos -- node run-local-testnet --force-restart --assume-yes
cargo run -p aptos-faucet-service -- run-simple --key ~/.aptos/testnet/mint.key --node-url http://127.0.0.1:8080 --chain-id TESTING
```
```
pnpm test
```

I also checked that the publishing stuff works.

### CODE REVIEW

File: crates/aptos-faucet/ts-client/src/models/RejectionReasonCode.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* istanbul ignore file */
+/* tslint:disable */
+/* eslint-disable */
+
+export enum RejectionReasonCode {
+    HEHE = 'Hehe',
```
Comment (0xmaayan) [inline crates/aptos-faucet/ts-client/src/models/RejectionReasonCode.ts:6]: :no_mouth:

File: crates/aptos-faucet/ts-client/src/services/GeneralService.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Check API health
+     * Basic endpoint that always returns Ok for health.
+     * @returns string
+     * @throws ApiError
+     */
+    public root(): CancelablePromise<string> {
+        return this.httpRequest.request({
+            method: 'GET',
+            url: '/',
```
Comment (0xmaayan) [inline crates/aptos-faucet/ts-client/src/services/GeneralService.ts:34]: why not `/health`?

**Review by 0xmaayan — APPROVED** [review]

File: crates/aptos-faucet/ts-client/src/models/RejectionReasonCode.ts (hunk 1 above)
Comment (banool) [reply crates/aptos-faucet/ts-client/src/models/RejectionReasonCode.ts:6]: lol

File: crates/aptos-faucet/ts-client/src/services/GeneralService.ts (hunk 2 above)
Comment (banool) [reply crates/aptos-faucet/ts-client/src/services/GeneralService.ts:34]: In GCP and other cloud deployments the default health check endpoint is `/` so it made sense.

File: crates/aptos-faucet/ts-client/src/models/RejectionReasonCode.ts (hunk 1 above)
Comment (banool) [reply crates/aptos-faucet/ts-client/src/models/RejectionReasonCode.ts:6]: lol yeah there is actually a good reason for this, I'll tell you later

**Review by gregnazario — APPROVED** [review]

**Pull request merged**